bot/src/user/user.controller.ts

```diff
@@ -17,13 +23,35 @@ export class UserController {
 
     const user = await this.userService.getUserByChatId(data.user.id);
 
+    //Fire and forget
+    this.userService.updateUsername(data.user.id)
+
     return {
       init_data: data,
       token: this.userService.getJwtToken(user.chat_id),
       is_connected: !!user.access_token
     }
   }
 
+  @Auth()
+  @Get('/friends')
+  async getFriends(@Me() user: User): Promise<FriendsDto> {
+    const friends = await this.userService.getFriendsByIds(user.friends)
+    const friendList: Friend[] = []
+    for await (const friend of friends) {
+      const info = await this.botService.getChat(friend.chat_id)
+      const avatar_url = info.photo ? await this.botService.getFileLink(info.photo.small_file_id) : undefined
```

Comment:
transfer checking fileId to getFileLink(), this should return null if it doesn't have fileId